proposals/0028-render-content.md

```diff
@@ -0,0 +1,300 @@
+			} else {
+				discoveredStyles.add(currentModule);
+			}
+		}
+	}
+}
+```
+
+## Adding `src/content` resources from `renderContent`
+
+Once we’ve pulled our `src/content` resources for later, we need to inject these resources onto the page. Today, Astro frontmatter can reach for the `$$result` object to access this.
+
+Our draft implementation injects the following into any **Astro** file that explicitly imports `renderContent`:
+
+```tsx
+import { renderContent } from '.astro';
+/* Injected */ import { renderContent as $$renderContent } from '.astro';
+...
+const $$stdin = $$createcomponent(async ($$result, $$props, $$slots) => {
+	const Astro = $$result.createAstro($$Astro, $$props, $$slots);
+	Astro.self = $$stdin;
+	/* Injected */ let renderContent = $$renderContent.bind($$result);
+	const userFrontmatterStuff...	
+}));
+```
+
+Two important pieces here:
+
+- We inject a separate `$$renderContent` import to modify what variables it may access.
+- We declare a local variable in the compiled frontmatter, ensuring the variable name matches the named `renderContent` import from the user. This will shadow the user import so we can `bind` the result object for `renderContent` to access.
+
+With this complete, `renderContent` can now update resource sets using, say. `this.links.add(...)`
+
+## A new `renderContentMap`
+
+The Content Schemas proposal [presented a new manifest](https://github.com/withastro/rfcs/blob/content-schemas/proposals/0027-content-schemas.md) generated from `src/content`, stored in a `.astro` directory as a cache. We expect `renderContent` to add a lazy `import.meta.glob` call (see background) so we can avoid loading each module until used.
+
+```tsx
+export const renderContentMap = import.meta.glob('src/content/**/*.{md,mdx}', { query: { SPECIAL_FLAG: true } });
+```
+
+# `renderContent` API reference
+
+- **************Param:************** `entry: ReturnType<fetchContent> | ReturnType<fetchContent>['id']`
+    - Either a complete `fetchContent` return type, or the ID of the entry to render. The ID type is a union of all valid IDs in your `src/content` directory (not a generic string) for better type checking.
+- ******************Returns: `{ Content: AstroComponentFactory }`**
+    - A `Content` component for use in Astro or MDX files
+
+## Full usage breakdown
+
+```tsx
+---
+import { fetchContent, renderContent } from '.astro';
+
+// Option 1: renderContent by fetchContent return value
+const firstNewsletter = await fetchContent('newsletter', 'first.md');
+const { Content } = await renderContent(firstNewsletter);
+
+// Option 2: renderContent by ID
+// Useful when the full `fetchContent` return object is not available
+const { Content } = await renderContent('newsletter/first.md');
+// Or, if you just have an entry ID from a `fetchContent` call:
+const { id, slug } = await fetchContent('newsletter', 'first.md');
+const { Content } = await renderContent(id);
+---
+
+<Content />
+```
+
```

Comment:
This proposal is missing these sections from the template:

* Testing strategy
* Drawbacks
* Adoption Strategy

Reply:
I listed some of the drawbacks I see in this comment: https://github.com/withastro/rfcs/pull/373#issuecomment-1314300882